File: dashboard/cores.py
import streamlit as st

# Paleta categorica em ordem fixa, validada para daltonismo nessa ordem
# (separacao entre vizinhas) e contra o fundo branco do dashboard.
PALETA = ["#2a78d6", "#eb6834", "#1baf7a", "#eda100", "#e87ba4", "#008300", "#4a3aa7", "#e34948"]


def atribuir(ufs: tuple[str, ...], atribuidas: dict[str, str]) -> dict[str, str]:
    # A cor acompanha a UF, nao a posicao dela na selecao: uma UF fica com a cor
    # que recebeu ao entrar no filtro, e tirar outra UF nao repinta as demais.
    for uf in list(atribuidas):
        if uf not in ufs:
            del atribuidas[uf]
    livres = [cor for cor in PALETA if cor not in atribuidas.values()]
    for uf in ufs:
        if uf not in atribuidas:
            atribuidas[uf] = livres.pop(0)
    return {uf: atribuidas[uf] for uf in ufs}


def por_uf(ufs: tuple[str, ...]) -> dict[str, str]:
    return atribuir(ufs, st.session_state.setdefault("cores_uf", {}))

File: web/src/lib/cores.ts
// Mesma paleta do dashboard Streamlit (dashboard/cores.py), validada para
// daltonismo nesta ordem. Um estado tem a mesma cor nos dois dashboards.
export const PALETA = [
  "#2a78d6",
  "#eb6834",
  "#1baf7a",
  "#eda100",
  "#e87ba4",
  "#008300",
  "#4a3aa7",
  "#e34948",
];

export const VERDE_MAPA: [string, string] = ["#eaf4ea", "#14532d"];

export function atribuir(
  ufs: string[],
  atribuidas: Record<string, string>,
): Record<string, string> {
  // A cor acompanha a UF, nao a posicao dela na selecao: tirar uma UF do filtro
  // nao repinta as demais.
  for (const uf of Object.keys(atribuidas)) {
    if (!ufs.includes(uf)) delete atribuidas[uf];
  }
  const usadas = new Set(Object.values(atribuidas));
  const livres = PALETA.filter((cor) => !usadas.has(cor));
  for (const uf of ufs) {
    if (!atribuidas[uf]) {
      const cor = livres.shift();
      if (cor) atribuidas[uf] = cor;
    }
  }
  return Object.fromEntries(ufs.filter((uf) => atribuidas[uf]).map((uf) => [uf, atribuidas[uf]]));
}
